ecp5/arch.cc

```diff
@@ -855,7 +855,7 @@ bool Arch::getCellDelay(const CellInfo *cell, IdString fromPort, IdString toPort
         std::string fn = fromPort.str(this), tn = toPort.str(this);
         if (fn.size() > 1 && (fn.front() == 'A' || fn.front() == 'B') && std::isdigit(fn.at(1))) {
             if (tn.size() > 1 && tn.front() == 'P' && std::isdigit(tn.at(1)))
-                return getDelayFromTimingDatabase(id_MULT18X18D_REGS_NONE, id(std::string("") + fn.front()), id_P,
+                return getDelayFromTimingDatabase(cell->multInfo.timing_id, id(std::string("") + fn.front()), id_P,
```

Comment:
This won't function correctly, clock-output and setup/hold times are accessed via getPortClockingInfo, not getCellDelay. For the latter, you will also need to query the database using getSetupHoldFromTimingDatabase. You will also need to update getPortTimingClass to report the DSP inputs/outputs as registered when the registers are enabled.

Reply:
Is it safe to still leave this as the same timing DB used elsewhere? Or is there a reason that it must be the REGS_NONE version?

Reply:
If a port is registered, then in fact getCellDelay should be returning false as it is only for combinational paths.

Reply:
Updated to return false early if the mult block is clocked